### PR TITLE
[core] Enable opaque pass for background layer. fix heatmap+backgroud test

### PR DESCRIPTION
**Pull request by astojilj**
Follow the approach from mapbox-gl-js for enabling opaque pass for background layer (same as for fill layer).

Fix combinations/heatmap-translucent--background-opaque render test.

### CODE REVIEW

File: src/mbgl/renderer/layers/render_background_layer.cpp [hunk 1]
```diff
@@ -37,8 +37,14 @@ void RenderBackgroundLayer::evaluate(const PropertyEvaluationParameters &paramet
...
         unevaluated.evaluate(parameters));
 
-    passes = properties->evaluated.get<style::BackgroundOpacity>() > 0 ? RenderPass::Translucent
-                                                                       : RenderPass::None;
+    passes = properties->evaluated.get<style::BackgroundOpacity>() == 0.0f
+        ? RenderPass::None
+        : (!unevaluated.get<style::BackgroundPattern>().isUndefined()
+           || properties->evaluated.get<style::BackgroundOpacity>() < 1.0f
+           || properties->evaluated.get<style::BackgroundColor>().a < 1.0f)
+        ? RenderPass::Translucent
```
Comment (pozdnyakov) [inline src/mbgl/renderer/layers/render_background_layer.cpp]:
```
if (properties->evaluated.get<style::BackgroundOpacity>() == 0.0f) {
passes = ..
} else if (...) {
passes = ...
} else {
passes = ...
}
```

File: src/mbgl/renderer/layers/render_background_layer.cpp [hunk 2]
```diff
@@ -118,6 +126,12 @@ void RenderBackgroundLayer::render(PaintParameters& parameters) {
             );
         }
     } else {
+        auto renderPass = (evaluated.get<BackgroundColor>().a >= 1.0f
```
Comment (pozdnyakov) [inline src/mbgl/renderer/layers/render_background_layer.cpp]: `allowedPass` ?

File: src/mbgl/renderer/layers/render_background_layer.cpp (hunk 1 above)
Comment (astojilj) [reply src/mbgl/renderer/layers/render_background_layer.cpp]: Actually, prefer original.

File: src/mbgl/renderer/layers/render_background_layer.cpp (hunk 2 above)
Comment (astojilj) [reply src/mbgl/renderer/layers/render_background_layer.cpp]: -> backgroundRenderPass, to be consistent with fill implementation.

File: src/mbgl/renderer/layers/render_background_layer.cpp (hunk 1 above)
Comment (pozdnyakov) [reply src/mbgl/renderer/layers/render_background_layer.cpp]: Up to you, IMO double ternary operator affects readability

**Pull request merged**